Fix read length plot for limited variation data.

whitwham opened the pull request:
Fixes #2068 where the limited length variations were causing the x range to be wrong.  Some other changes to quiet warnings about y ranges.

whitwham committed:
Fix read length plot for limited variation data.

daviesrob commented on misc/plot-bamstats:
```diff
@@ -1231,6 +1231,44 @@ sub plot_indel_cycles
+
+    my $nf;
+
+    if ($fract_num <= 1.0) {
```
This could be simplified a bit by importing `ceil` from `POSIX`, then doing `my $nf = ceil($fract_num);`.

I don't know if `$num` can ever be zero, which would cause trouble in `log()`?  I suspect not, but you could make sure with `if ($num < 1) { return 1; }`.

whitwham replied:
Yeah, that was left over from a more complicated function I ended up not liking the results from.  I'll change it.

daviesrob reviewed (commented):
Looks like a good fix for the read length plot, although in testing I found it's still possible to break `plot-bamstats` elsewhere by not giving it much data. E.g.:

```
mkdir /tmp/stats
./samtools stats test/dat/view.002.sam > /tmp/view.002.stats
./misc/plot-bamstats -p /tmp/stats/view.002 /tmp/view.002.stats 
Warning: empty x range [1:1], adjusting to [0.99:1.01]
Warning: empty x range [1:1], adjusting to [0.99:1.01]
Warning: empty x range [2:2], adjusting to [1.98:2.02]

plot '-' smooth csplines with lines lc 1 title 'First fragments' 
     ^
"/tmp/stats/view.002-gc-content.gp", line 12: Can't calculate splines, need at least 3 points

The command exited with non-zero status 256:
	gnuplot /tmp/stats/view.002-gc-content.gp

```
This may be something to look at in the future.

Meanwhile, https://github.com/samtools/samtools/issues/2068#issue-2340100500 notes that the read length plot doesn't appear on the html page.  It doesn't look like it should be too hard to add...

whitwham committed:
Simplified rounding as requested in review.

Added read-length to html page.

Changed file suffix from read_length to read-length to match the other file names.
